### PR TITLE
Engdocs 1871

### PR DESCRIPTION
**Pull request by aevesdocker**
Closes https://github.com/docker/docs/issues/18750
Closes https://github.com/docker/docs/issues/18666

### CODE REVIEW

File: content/compose/reference/_index.md [hunk 1]
```diff
@@ -36,7 +23,7 @@ Define and run multi-container applications with Docker.
 Options:
       --ansi string                Control when to print ANSI control characters ("never"|"always"|"auto") (default "auto")
       --compatibility              Run compose in backward compatibility mode
```
Comment (dvdksn) [inline content/compose/reference/_index.md:25]: The `--dry-run` flag is missing

File: content/compose/reference/_index.md [hunk 2]
```diff
@@ -36,7 +23,7 @@ Define and run multi-container applications with Docker.
...
-      --env-file stringArray       Specify an alternate environment file.
+      --env-file stringArray       Specify an alternate environment file
   -f, --file stringArray           Compose configuration files
       --parallel int               Control max parallelism, -1 for unlimited (default -1)
       --profile stringArray        Specify a profile to enable
```
Comment (dvdksn) [inline content/compose/reference/_index.md:29]: The `--progress` flag is missing

File: content/compose/reference/_index.md [hunk 3]
```diff
@@ -36,7 +23,7 @@ Define and run multi-container applications with Docker.
 Options:
       --ansi string                Control when to print ANSI control characters ("never"|"always"|"auto") (default "auto")
       --compatibility              Run compose in backward compatibility mode
-      --env-file stringArray       Specify an alternate environment file.
+      --env-file stringArray       Specify an alternate environment file
```
Comment (dvdksn) [inline content/compose/reference/_index.md:26]: This period is there in the cobra help string so we probably want to update it in the code as well

File: content/compose/reference/_index.md [hunk 4]
```diff
@@ -126,12 +113,12 @@ relative to the current working directory.
```
Comment (dvdksn) [inline content/compose/reference/_index.md:113]: unrelated to the changes here, but we could probably use an example here for reading the compose file from stdin with a heredoc

File: content/compose/reference/_index.md [hunk 5]
```diff
@@ -105,7 +92,7 @@ webapp:
     - "/data"
 ```
 
-If the `docker-compose.admin.yml` also specifies this same service, any matching
+If the `compose.admin.yml` also specifies this same service, any matching
 fields override the previous file. New values, add to the `webapp` service
 configuration.
```
Comment (dvdksn) [inline content/compose/reference/_index.md:97]: Not related to this change, but this description is inaccurate. Some fields are merged, not overridden. Labels, for example:

```yaml
# compose.yaml
services:
  webapp:
    build:
      context: .
      labels: 
        com.example.foo: "foo"
```

```yaml
# overrides.yaml
services:
  webapp:
    build:
      context: .
      labels: 
        com.example.name: "Alice"
```

```console
$ docker compose -f compose.yaml -f overrides.yaml build
$ docker image inspect compose-webapp --format json | jq '.[].Config.Labels'
{
  "com.docker.compose.project": "compose",
  "com.docker.compose.service": "webapp",
  "com.docker.compose.version": "2.23.0",
  "com.example.foo": "foo",
  "com.example.name": "Alice"
}
```

`buildx bake` has similar behavior and I've described how duplicate fields either merge with or override previous definitions in the bake file reference, see here: https://docs.docker.com/build/bake/reference/#file-format

File: content/compose/reference/_index.md [hunk 6]
```diff
@@ -126,12 +113,12 @@ relative to the current working directory.
 
 The `-f` flag is optional. If you don't provide this flag on the command line,
 Compose traverses the working directory and its parent directories looking for a
-`docker-compose.yml` and a `docker-compose.override.yml` file. You must supply
-at least the `docker-compose.yml` file. If both files are present on the same
+`compose.yml` and a `compose.override.yml` file. You must supply
+at least the `compose.yml` file. If both files are present on the same
```
Comment (dvdksn) [inline content/compose/reference/_index.md:117]: Both `docker-compose.yml` and `compose.yml` and corresponding override files are supported. Compose has some logic for deciding the lookup order of these files, which is what I think #18666 is asking for. How do these variants interact?

Again the bake file has a similar behavior so you can take a look at how I did it there: https://docs.docker.com/build/bake/reference/#file-format

**Review by dvdksn — CHANGES_REQUESTED** [review]
I have one specific comment related to the change, which is about file lookup order. The current change doesn't address the fact that you can have multiple compose filenames and they take presedence in different ways.

Then I had several comments on things unrelated to these changes but that I noticed in passing.

**Comment by aevesdocker** [comment]
Thanks @dvdksn. It's on my radar that this page needs some TLC. I've addressed your main blocking comment. The rest I'll do in a follow-up

**Review by dvdksn — APPROVED** [review]

**Pull request merged**